src/main/java/com/limechain/teavm/HttpRequest.java

```diff
@@ -1,8 +1,32 @@
 package com.limechain.teavm;
 
+import org.teavm.interop.Async;
+import org.teavm.interop.AsyncCallback;
 import org.teavm.jso.JSBody;
+import org.teavm.jso.JSFunctor;
+import org.teavm.jso.JSObject;
+import org.teavm.jso.core.JSError;
+
+import java.io.IOException;
 
 public class HttpRequest {
-    @JSBody(params = {"method", "url", "body"}, script = "return httpRequestSync(method, url, body);")
-    public static native String httpRequestSync(String method, String url, String body);
+    @JSFunctor
+    interface HttpRequestCallback extends JSObject {
+        void apply(JSError error, String response);
+    }
+
+    @Async
+    public static native String asyncHttpRequest(String method, String url, JSObject body);
+    private static void asyncHttpRequest(String method, String url, JSObject body, AsyncCallback<String> callback) {
```

Comment:
Is this called anywhere?

Reply:
In order to translate the js async operation to java sync method we need them both.

Reply:
![Uploading Screenshot 2024-08-14 at 17.37.45.png…]()

Reply:
![Screenshot 2024-08-14 at 17 37 45](https://github.com/user-attachments/assets/2e6088fd-d16e-4238-8632-dbbebe6516fe)